src/AhRulesBot/AhRulesBot/Infrastructure/ServiceCollectionExtensions/CardsServiceCollectionExtensions.cs

```diff
@@ -10,7 +11,15 @@ internal static class CardsServiceCollectionExtensions
     {
         internal static IServiceCollection AddCardsFile(this IServiceCollection services)
         {
-            return services.AddSingleton(x => JsonSerializer.Deserialize<List<Card>>(File.ReadAllText(x.GetService<AppConfig>().CardsFilePath)));
+            return services.AddSingleton(x =>
+            {
+                var data = JsonSerializer.Deserialize<List<Card>>(File.ReadAllText(x.GetRequiredService<AppConfig>().CardsFilePath));
+                if (data == null)
+                {
+                    throw new Exception("cards is null");
```

Comment:
if you want to implement proper config validation, I would suggest to use `IValidateOptions` 
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-5.0#use-di-services-to-configure-options
https://stackoverflow.com/questions/57135724/how-to-validate-configuration-settings-using-ivalidateoptions-in-asp-net-core-2